internal/services/streamanalytics/stream_analytics_job_resource.go

```diff
@@ -145,16 +145,6 @@ func resourceStreamAnalyticsJob() *pluginsdk.Resource {
 				Optional: true,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*schema.Schema{
-						"authentication_mode": {
```

Comment:
This is a breaking change? 

how about instead of removing it we mark it optional, default to connection string and then remove the invalid properties?

Reply:
Hi @katbyte , now `authentication_mode` here is actually not configurable for users. The only value it can take is `ConnectionString`. If any user set it as `Msi` or anything else, it will do nothing but emit an error message. That's why we would prefer to remove it from the schema.

Reply:
Can you put the removal of `authentication_mode` behind the 4.0 feature flag, and for now set this to optional, default to connection string and remove the invalid values as @katbyte suggested above?

Reply:
Changed as suggested. Will not deprecate `authentication_mode` in 4.0 for now since support for other modes may come up shortly.